File: C# Programming Basics January 22'/Basics - all the sc/Gold Mine.cs
using System;

namespace _06._Gold_Mine
{
    class Program
    {
        static void Main(string[] args)
        {
            //input
            int locationsCount = int.Parse(Console.ReadLine());
            double totalAverage = 0;

            for (int i = 1; i <= locationsCount; i++)
            {
                double expectedAvergaeSupply = double.Parse(Console.ReadLine());
                int daysForMining = int.Parse(Console.ReadLine());
                double totalMined = 0;

                for (int j = 1; j <= daysForMining; j++)
                {
                    double minedGold = double.Parse(Console.ReadLine());
                    totalMined += minedGold;


                }
                totalAverage = totalMined / daysForMining;
                if (totalAverage >= expectedAvergaeSupply)
                {
                    Console.WriteLine($"Good job! Average gold per day: {totalAverage:f2}.");

                }
                else
                {
                    Console.WriteLine($"You need {Math.Abs(expectedAvergaeSupply - totalAverage):f2} gold.");

                }
            }

        }
    }
}

File: Python/PythonBasic/Exam/gold_mine.py
locations_cnt = int(input())

for i in range(locations_cnt):
    expected_per_day = float(input())
    days_mining = int(input())
    total_mined = 0

    for j in range(days_mining):
        mined_gold_for_day = float(input())
        total_mined += mined_gold_for_day
    total_mined /= days_mining

    if total_mined >= expected_per_day:
        print(f"Good job! Average gold per day: {total_mined:.2f}.")
    else:
        print(f"You need {expected_per_day - total_mined:.2f} gold.")
